src/vs/workbench/contrib/chat/common/chatModel.ts

```diff
@@ -1585,6 +1591,7 @@ export class ChatModel extends Disposable implements IChatModel {
 		}
 
 		request.response.complete();
+		this._onDidChange.fire({ kind: 'completedRequest', request });
```

Comment:
We access `chatModelrequestInProgress` in `chatEditingEditorContextKeys.ts`.
However the `onDidChange` event doesn't trigger when the response is complete.
I found that this was the case in some cases for regular text editors as well.

More like a race condition.
In some cases the response complete variable is set before the session state is change.

Here onDidChange is never triggered, hence this is effectively the same as 
```const isRequestInProgress = chatModel?.requestInProgress === true```
Usage
```typescript

			const isRequestInProgress = chatModel
				? observableFromEvent(this, chatModel.onDidChange, () => chatModel.requestInProgress)
				: constObservable(false);
```

Reply:
Thank you. I came to the same conclusion and would have done the same change.